packages/block-editor/src/store/selectors.js

```diff
@@ -458,6 +458,37 @@ export const getBlockParents = createSelector(
 	( state ) => [ state.blocks.parents ]
 );
 
+/**
+ * Given a block client ID and a block name,
+ * returns the list of all its parents from top to bottom,
+ * filtered by the given name.
+ *
+ * @param {Object} state     Editor state.
+ * @param {string} clientId  Block from which to find root client ID.
+ * @param {string} blockName Block name to filter.
+ * @param {boolean} ascending Order results from bottom to top (true) or top to bottom (false).
+ *
+ * @return {Array} ClientIDs of the parent blocks.
+ */
+export const getBlockParentsByBlockName = (
+	state,
+	clientId,
+	blockName,
+	ascending = false
+) => {
+	const parents = getBlockParents( state, clientId, ascending );
+	return map(
```

Comment:
This selector will return new array on each call potentially causing performance issues, should we use createSelector?

Reply:
re-implemented here https://github.com/WordPress/gutenberg/pull/20057